src/main.rs

```diff
@@ -26,6 +26,9 @@ use vmm_sys_util::signal::block_signal;
 
 #[derive(Error, Debug)]
 enum Error {
+    #[cfg(feature = "amx")]
+    #[error("Failed to setup AMX: {0}")]
+    AmxEnable(String),
```

Comment:
This is a general comment but AMX should also be gated behind the `x86_64` arch feature, since it doesn't make any sense for `aarch64`.

Reply:
Ah good point, I was mostly using `tdx` as my model.